src/web/src/layouts/atm/components/CustomWithdrawAmount.tsx

```diff
@@ -17,11 +17,14 @@ interface Props {
   account: Account;
 }
 
-const formSchema = z.object({
-  amount: z.number({ coerce: true, message: locales.amount_required_number }).min(1),
-});
-
 const CustomWithdrawAmount: React.FC<Props> = ({ isWithdrawing, handleWithdraw, account }) => {
+  const formSchema = z.object({
+    amount: z
+      .number({ coerce: true, message: locales.amount_required_number })
+      .min(1, { message: locales.amount_greater_than_zero })
+      .max(account?.balance ?? 0, { message: locales.amount_greater_than_balance }),
+  });
```

Comment:
The form schema should be inside of a  `React.useMemo` with nothing in the dependency array because we don't ever want to recreate it when the component rerenders.

Reply:
I believe the dependency should be account?.balance as we validate maximum amount.